lib/std/os.zig

```diff
@@ -3237,3 +3237,21 @@ pub fn sched_yield() SchedYieldError!void {
         else => return error.SystemCannotYield,
     }
 }
+
+/// Set a socket's options.
+pub fn setsockopt(fd: fd_t, level: u32, optname: u32, optval: [*]const u8, optlen: socklen_t) !void {
```

Comment:
zigify your wrapper a bit more: take an argument `opt: []const u8`  and extract out the `ptr` and `len`